doc/api/brotli.md

```diff
@@ -0,0 +1,404 @@
+stream:
+
+```js
+const compress = new brotli.Compress();
+const fs = require('fs');
+const inp = fs.createReadStream('input.txt');
+const out = fs.createWriteStream('input.txt.gz');
+
+inp.pipe(compress).pipe(out);
+```
+
+It is also possible to compress or decompress data in a single step:
+
+```js
+const input = '.................................';
+brotli.compress(input, (err, buffer) => {
+  if (!err) {
+    console.log(buffer.toString('base64'));
+  } else {
+    // handle error
+  }
+});
+
+const buffer = Buffer.from('GyAA+CVcgowA4A==', 'base64');
+brotli.decompress(buffer, (err, buffer) => {
+  if (!err) {
+    console.log(buffer.toString());
+  } else {
+    // handle error
+  }
+});
+```
+
+## Threadpool Usage
+
+Note that all brotli APIs except those that are explicitly synchronous use
+libuv's threadpool, which can have surprising and negative performance
+implications for some applications, see the [`UV_THREADPOOL_SIZE`][]
+documentation for more information.
+
+## Compressing HTTP requests and responses
+
+The `brotli` module can be used to implement support for the `br`
+content-encoding mechanism defined by
+[HTTP](https://tools.ietf.org/html/rfc7230#section-4.2).
+
+The HTTP [`Accept-Encoding`][] header is used within an http request to identify
+the compression encodings accepted by the client. The [`Content-Encoding`][]
+header is used to identify the compression encodings actually applied to a
+message.
+
+The examples given below are drastically simplified to show the basic concept.
+Using `brotli` encoding can be expensive, and the results ought to be cached.
+
+```js
+// client request example
+const brotli = require('brotli');
+const http = require('http');
+const fs = require('fs');
+const request = http.get({ host: 'example.com',
+                           path: '/',
+                           port: 80,
+                           headers: { 'Accept-Encoding': 'br' } });
+request.on('response', (response) => {
+  const output = fs.createWriteStream('example.com_index.html');
+
+  switch (response.headers['content-encoding']) {
+    case 'br':
+      response.pipe(new brotli.Decompress()).pipe(output);
+      break;
+    default:
+      response.pipe(output);
+      break;
+  }
+});
+```
+
+```js
+// server example
+// Running a brotli operation on every request is quite expensive.
+// It would be much more efficient to cache the compressed buffer.
+const brotli = require('brotli');
+const http = require('http');
+const fs = require('fs');
+http.createServer((request, response) => {
+  const raw = fs.createReadStream('index.html');
+  let acceptEncoding = request.headers['accept-encoding'];
+  if (!acceptEncoding) {
+    acceptEncoding = '';
+  }
+
+  // Note: This is not a conformant accept-encoding parser.
+  // See https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.3
+  // Note: A quality of 4 is a good balace of speed and quality.
```

Comment:
![](https://memegenerator.net/img/instances/62044377/there-are-how-shall-one-say-too-many-notes.jpg)

Reply:
I believe «Note: » prefixes were removed from the docs earlier in #18592.
Are those needed here?